AdobeStockClient/Model/Client.php

```diff
@@ -227,6 +240,64 @@ public function getQuotaConfirmationMessage(int $contentId): string
         return $this->getLicenseInfo($contentId)->getPurchaseOptions()->getMessage();
     }
 
+    /**
+     * Performs image license request to Adobe Stock APi
+     *
+     * @param int $contentId
+     * @return License
+     * @throws IntegrationException
+     * @throws \AdobeStock\Api\Exception\StockApi
+     */
+    public function licenseImage(int $contentId): License
+    {
+        $licenseRequest = $this->getLicenseRequest($contentId);
+        $license = $this->getConnection()->getContentLicense($licenseRequest, $this->getAccessToken());
+
+        return $license;
+    }
+
+    /**
+     * Returns download URL for a licensed image
+     *
+     * @param int $contentId
+     * @return string
+     * @throws IntegrationException
+     * @throws \AdobeStock\Api\Exception\StockApi
+     */
+    public function getImageDownloadUrl(int $contentId): string
+    {
+        $licenseRequest = $this->getLicenseRequest($contentId);
+
+        return $this->getConnection()->downloadAssetUrl($licenseRequest, $this->getAccessToken());
+    }
+
+    /**
+     * Convert a stock file object to a document object
+     *
+     * @param StockFile $file
+     * @return DocumentInterface
+     * @throws IntegrationException
+     */
+    private function convertStockFileToDocument(StockFile $file): DocumentInterface
```

Comment:
`\Magento\AdobeStockClient\Model\StockFileToDocument::convert` should be used instead